File: Nodejs/K06.ts
import { IExecutable } from './IExecutable';
import * as readline from 'readline';
import { format } from 'util';

export class K06 extends IExecutable {

    Question1(rl?: readline.ReadLine): Promise<string> {
        try {
            let ages = new Array<number>();
            let count = 0;

            let q1 = () => {
                return new Promise((resolve, reject) => {
                    rl.question(format("%d人目の年齢を入力して下さい：", count + 1), (value) => {
                        ages[count] = Number.parseInt(value);
                        resolve();
                    });
                });
            }

            return new Promise(async (resolve, reject) => {
                for (var i = 0; i < 5; i++) {
                    count = i;
                    await q1();
                }
                let average = ages.reduce((prev, cur, index, arr) => {
                    return prev + cur;
                }) / ages.length;
                console.log(format("%d人の平均年齢は%dです。", ages.length, average));
                resolve();
            });
        } catch (e) {
            console.error(e);
        }
    }

    Question2(rl?: readline.ReadLine): Promise<string> {
        for (var i = 0; i < 9; i++)
        {
            for (var j = 0; j < i + 1; j++)
            {
                process.stdout.write("*");
            }
            console.log();
        }
        console.log();

        for (var i = 9; i > 0; i--)
        {
            for (var j = 0; j < i; j++)
            {
                process.stdout.write("*");
            }
            console.log();
        }
        console.log();

        for (var i = 9; i > 0; i--)
        {
            for (var j = 0; j < i - 1; j++)
            {
                process.stdout.write(" ");
            }

            for (var k = 9; k > i - 1; k--)
            {
                process.stdout.write("*");
            }
            console.log();
        }

        return new Promise((resolve, reject) => {
            resolve();
        });
    }

    Question3(rl?: readline.ReadLine): Promise<string> {
        let total = 370;
        let count = 0;

        for (var i = 0; i <= Math.trunc(total / 100); i++) {
            for (var j = 0; j <= Math.trunc(total / 50); j++) {
                for (var k = 0; k <= Math.trunc(total / 10); k++) {
                    if (100 * i + 50 * j + 10 * k == total) {
                        console.log(format("10円の硬貨%d枚 50円の硬貨%d枚 100円の硬貨", k, j, i));
                        count++;
                    }
                }
            }
        }

        console.log("\n以上" + count + "通りを発見しました。");
        return new Promise((resolve, reject) => {
            resolve();
        });
    }

    Question4(rl?: readline.ReadLine): Promise<string> {

        process.stdout.write("\t|\t");

        for (var i = 1; i < 10; i++)
        {
            process.stdout.write(i + "\t");
        }
        console.log();

        for (var i = 0; i < 76; i++)
        {
            process.stdout.write("-");
        }
        console.log();

        for (var i = 1; i < 10; i++)
        {
            process.stdout.write(i + "\t|\t");
            for (var j = 1; j < 10; j++)
            {
                process.stdout.write((i * j) + "\t");
            }
            console.log();
        }

        return new Promise((resolve, reject) => {
            resolve();
        });
    }

}

File: Nim/K06.nim
import os
from strutils import parseInt
import Golem, IExecutable

type K06* = ref object of IExecutable

method question1*(this: K06) =
    var ages: seq[int]
    for i in 1..5:
        stdout.write i, "人目の年齢を入力して下さい："
        ages.add(parseInt(readLine(stdin)))

    var average = 0.0
    for i in ages.items:
        average += i.toFloat
    average /= ages.len.toFloat
    echo ages.len, "人の平均年齢は", average, "です。"

method question2*(this: K06) =
    for i in 1..9:
        for j in 1..i:
            stdout.write "*"
            stdout.flushFile
        echo ""
    echo ""

    for i in countdown(9, 1):
        for j in 1..i:
            stdout.write "*"
            stdout.flushFile
        echo ""
    echo ""

    for i in countdown(9, 1):
        for j in countup(1, i - 1):
            stdout.write " "
            stdout.flushFile
        
        for k in countdown(9, i):
            stdout.write "*"
            stdout.flushFile
        echo ""


method question3*(this: K06) =
    var total = 370
    var count = 0
    for i in countup(0, total div 100):
        for j in countup(0, total div 50):
            for k in countup(0, total div 10):
                if 100 * i + 50 * j + 10 * k == total:
                    echo "10円の硬貨", k, "枚 50円の硬貨", j, "枚 100円の硬貨", i, "枚"
                    inc(count)
    echo "\n以上", count, "通りを発見しました。"                
        
    

method question4*(this: K06) =
    stdout.write "\t|\t"
    stdout.flushFile

    for i in 1..9:
        stdout.write i, "\t"
        stdout.flushFile
    echo ""

    for i in 1..85:
        stdout.write "-"
        stdout.flushFile
    echo ""

    for i in 1..9:
        stdout.write i, "\t|\t"
        stdout.flushFile
        for j in 1..9:
            stdout.write i * j, "\t"
        echo ""